STM32Cube/Tasks/flight_phase.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * flight_phase.c
+ *
+ *  Created on: May 17, 2024
+ *      Author: joedo
+ */
+
+#include "flight_phase.h"
+#include "millis.h"
+
+#define BOOST_TIME_MS 15000 //time from inj valve open to motor bunout
+#define RECOVERY_DELAY_MS 30000 //time from inj valve open to recovery deployment
+
+enum FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
+float injOpenTime;
+float deploymentTime;
+
+EventGroupHandle_t flightPhaseEventsHandle = NULL;
+#define INJ_OPEN (xEventGroupGetBits(flightPhaseEventsHandle) && INJ_OPEN_BIT)
+#define RECOVERY_DEPLOYMENT (xEventGroupGetBits(flightPhaseEventsHandle) && RECOVERY_DEPLOYMENT_BIT)
+
+
+void flightPhaseInit()
+{
+	flightPhaseEventsHandle = xEventGroupCreate();
+	xEventGroupClearBits(flightPhaseEventsHandle, 0xFFFF); //clear out the enitire group since API doesn't specify if values are initialized to 0
+}
+
+void flightPhaseTask(void * argument)
+{
+	while(1)
+	{
+		if(flight_state == PHASE_PRELAUNCH && INJ_OPEN)
+		{
+			injOpenTime = millis();
+			flight_state = 	PHASE_BOOST;
+		}
+
+		else if(flight_state == PHASE_BOOST && (millis() - injOpenTime) > BOOST_TIME_MS)
```

Comment:
Is it better to specify that the state that it is coming from is boost or just have the conditions that cause it to be in coast set it to be in coast regardless of where it came from. 

Also would it not make more sense to switch this else if with the other one, in terms of order, just because conventionally the more restrictive condition (in this case in terms of elapsed time) comes before the less restrictive one?

Reply:
Well yeah
We can't expect millis() - injOpenTime to yield a valid result unless we know we are in the boost phase and therefore injOpenTime has a sensible value.

Reply:
yup that makes sense since thats where u start the reference for the timer